Structure_Engine/Query/Geometry3D.cs

```diff
@@ -99,8 +102,62 @@ public static IGeometry Geometry3D(this Panel panel, bool onlyCentralSurface = f
                 compositeGeometry.Elements.AddRange(externalEdgesExtrusions);
 
                 return compositeGeometry;
+
             }
         }
+
+        /***************************************************/
+
+        [Description("Gets the BH.oM.Geometry.Extrusion out of the Pile as its Geometry3D.")]
+        [Input("pile", "The input Pile to get the Geometry3D out of, i.e.its extrusion with its cross section along its centreline.")]
+        [Output("Mesh", "The mesh defining the 3D geometry of the Pile.")]
+        public static IGeometry Geometry3D(this Pile pile)
+        {
+            return Create.Bar((Line)pile.Geometry(), pile.Section, pile.OrientationAngle).Geometry3D();
```

Comment:
I think you need to add an `Extrude` method here for `Pile` - creating a `Bar` from the `Pile` to generate the extrusion is not acceptable.

Reply:
This can be left for now, I have raised an issue here to fix it next milestone:
https://github.com/BHoM/BHoM_Engine/issues/3352